indigo_social/forms.py

```diff
@@ -37,8 +43,10 @@ def clean_twitter_username(self):
             return twitter_username
 
     def clean_username(self):
-        if self.cleaned_data['username']:
-            return generate_unique_username([self.cleaned_data['username']])
+        username = self.cleaned_data['username']
+        if username in [u.username for u in User.objects.all()]:
```

Comment:
I think this will raise an error even if it's the user's current username. How about:

```python
if User.objects.filter(username=username).exclude(pk=self.request.user.pk).exists():
```

I'm still suspicious that we even have to do this.